src/vs/base/common/comparers.ts

```diff
@@ -6,30 +6,168 @@
 import { sep } from 'vs/base/common/path';
 import { IdleValue } from 'vs/base/common/async';
 
-const intlFileNameCollator: IdleValue<{ collator: Intl.Collator, collatorIsNumeric: boolean }> = new IdleValue(() => {
-	const collator = new Intl.Collator(undefined, { numeric: true, sensitivity: 'base' });
+// When comparing large numbers of strings, such as in sorting large arrays, is better for
+// performance to create an Intl.Collator object and use the function provided by its compare
+// property than it is to use String.prototype.localeCompare()
+
+// A collator with numeric sorting enabled.
+const intlFileNameCollatorNumeric: IdleValue<{ collator: Intl.Collator }> = new IdleValue(() => {
+	const collator = new Intl.Collator(undefined, { numeric: true });
+	return {
+		collator: collator
+	};
+});
+
+// A collator with numeric sorting enabled, and sensitivity to accents but not case.
+const intlFileNameCollatorNumericCaseInsenstive: IdleValue<{ collator: Intl.Collator }> = new IdleValue(() => {
+	const collator = new Intl.Collator(undefined, { numeric: true, sensitivity: 'accent' });
 	return {
-		collator: collator,
-		collatorIsNumeric: collator.resolvedOptions().numeric
+		collator: collator
 	};
 });
 
-export function compareFileNames(one: string | null, other: string | null, caseSensitive = false): number {
-	const a = one || '';
-	const b = other || '';
-	const result = intlFileNameCollator.getValue().collator.compare(a, b);
+// A collator with numeric sorting disabled.
+const intlFileNameCollator: IdleValue<{ collator: Intl.Collator }> = new IdleValue(() => {
+	const collator = new Intl.Collator(undefined, { numeric: false });
+	return {
+		collator: collator
+	};
+});
 
-	// Using the numeric option in the collator will
-	// make compare(`foo1`, `foo01`) === 0. We must disambiguate.
-	if (intlFileNameCollator.getValue().collatorIsNumeric && result === 0 && a !== b) {
-		return a < b ? -1 : 1;
+// A collator with numeric sorting disabled and sensitivity to accents but not case.
+const intlFileNameCollatorCaseInsensitive: IdleValue<{ collator: Intl.Collator }> = new IdleValue(() => {
+	const collator = new Intl.Collator(undefined, { numeric: false, sensitivity: 'accent' });
+	return {
+		collator: collator
+	};
+});
+
+// Export compareFileNamesNumeric by its old name too for backwards compatibility.
+// Also keeping a placeholder parameter for the same reason. The parameter was sometimes supplied but was
+// not used in the original code.
+export { compareFileNamesNumeric as compareFileNames };
```

Comment:
Since we seem to be supporting the old name, why are there changes in this PR to change to `compareFileNamesNumeric`?

Reply:
I don't really want to support the old name - but I wasn't sure how to safely change it? If I don't support the old name there are problems in CI building the stand-alone editor.

I've updated the code so that instead of just exporting an alias (which didn't work with tree-shaking) I export a compareFileNames function that calls compareFileNamesNumeric - so now the CI checks pass at least.